build.gradle

```
@@ -221,12 +218,13 @@ htmlDependencyReport {
 // 'jacocoTestReport' is provided by the 'jacoco' plugin
 // docs: https://docs.gradle.org/current/userguide/jacoco_plugin.html
 jacocoTestReport {
-    dependsOn test
```

Why did you remove the dependency here? The test report task is useless without the test run or am I missing something?

---

I removed it, because: If the test fails, the report isn't generated at all.
But if I run it separately, it works. This is, why I adjusted the Taskfile tasks.

---

I tried many things to solve that. Even, I asked Nitin and Ronak, if they know a solution...

---

You can add a gradle property to ignoreFailures

```properties
# in gradle.properties
...
testIgnoreFailures=false
```

then use it:

```groovy
test {
    useJUnitPlatform()
    ignoreFailures = testIgnoreFailures.toBoolean()
}
```

Finally trigger it on the cli

```bash
./gradlew clean test -PtestIgnoreFailures=true
```

BUT shouldn't the pipeline fail if the tests fail?

---

In general, the pipeline should fail.
So, if your suggestion works, I will undo some changes and let the pipeline fail.